src/frontend/Semantic_check.ml

```diff
@@ -591,9 +591,22 @@ and semantic_check_expression cf ({emeta; expr} : Ast.untyped_expression) :
       Validate.(liftA3 tuple3 pe te fe >>= semantic_check_ternary_if emeta.loc)
   | BinOp (e1, op, e2) ->
       let le = semantic_check_expression cf e1
-      and re = semantic_check_expression cf e2 in
+      and re = semantic_check_expression cf e2
+      and warn_int_division (x, y) =
+        match (x.emeta.type_, y.emeta.type_, op) with
+        | UInt, UReal, Divide | UReal, UInt, Divide ->
+            Fmt.pr
+              "Info: integer division implicitly rounds to integer. Found int \
+               division: %a / %a\n\
+              \      Positive values rounded down, negative values rounded up \
```

Comment:
May look nicer here to use OCaml's multi-line strings, which are anything put between `{|` and `|}`.